MySQLdb/_mysql.c

```diff
@@ -486,6 +488,21 @@ _mysql_ConnectionObject_Initialize(
     }
     Py_BEGIN_ALLOW_THREADS ;
     self->open = 1;
+
+    #ifdef MARIADB_BASE_VERSION
+    if (ssl_mode) {
+        if (strcasecmp(ssl_mode, "PREFERRED") != 0)
+        {
+            int enforce_tls= 0;
+            if (strcasecmp(ssl_mode, "REQUIRED") == 0)
+                enforce_tls = 1;
+            #ifdef MYSQL_OPT_SSL_ENFORCE
```

Comment:
Yeah, that is precisely the problem, because MariaDB forked from MySQL and apparently so have got their own way to do things. And these things are no longer consistent with MySQL.